src/main/java/frc/robot/subsystems/CDSSubsystem.java

```diff
@@ -8,51 +8,86 @@
 import frc.robot.Constants;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import frc.robot.common.hardware.MotorController;
+import edu.wpi.first.wpilibj.DigitalInput;
 
 /** Add your docs here. */
 public class CDSSubsystem extends SubsystemBase {
   // Put methods for controlling this subsystem
   // here. Call these from Commands.
   private MotorController CDSBeltController;
-  private MotorController CDSWheelControllerOne;
-  private MotorController CDSWheelControllerTwo;
-  
+  private MotorController CDSMotorController2;
+
+  private DigitalInput intitalBallSensor;
+  private DigitalInput middleBallSensor;
+  private DigitalInput finalBallSensor;
+  private int ballCount = 0;
+ 
+  //TODO: Figure out what CDS motor is for
   public CDSSubsystem() {
-    CDSBeltController = new MotorController("CDS Motor", Constants.kCDSMotorThreeID);
-    CDSWheelControllerOne =  new MotorController("Main CDS Wheel Controller", Constants.kCDSMotorFourID);
-    CDSWheelControllerTwo = new MotorController("Follows CDS Wheel Controller", Constants.kCDSMotorFiveID);
-    
-    CDSWheelControllerTwo.getSparkMax().follow(CDSWheelControllerOne.getSparkMax());
+    CDSBeltController = new MotorController("CDS Motor", Constants.CDSBeltID);
+    CDSMotorController2 = new MotorController("CDS Motor 2", Constants.CDSMotorID);
+   
+    intitalBallSensor = new DigitalInput(Constants.ballSensorChannel);
+    middleBallSensor = new DigitalInput(Constants.ballSensorChannel);
+    finalBallSensor = new DigitalInput(Constants.ballSensorChannel);
   }
 
-  public void HopperSwitch(boolean on) {
+  public void CDSSwitch(boolean on) {
     if (on) {
-      double beltSmartSpeed = SmartDashboard.getNumber("Belt Speed", Constants.kCDSBeltSpeed);
-      double wheelSmartSpeed = SmartDashboard.getNumber("Wheel Speed", Constants.kCDSWheelSpeed);
-
-      CDSBeltController.getSparkMax().set(beltSmartSpeed);
-      CDSWheelControllerOne.getSparkMax().set(wheelSmartSpeed);
-      SmartDashboard.putNumber("CDS Belt Speed", Constants.kCDSBeltSpeed);
-      SmartDashboard.putNumber("CDS Wheel Speed", Constants.kCDSWheelSpeed);
+      CDSBeltController.getSparkMax().set(Constants.CDSBeltSpeed);
+      SmartDashboard.putNumber("CDS Belt Speed", Constants.CDSBeltSpeed);
     } else {
       CDSBeltController.getSparkMax().set(0.0);
-      CDSWheelControllerOne.getSparkMax().set(0.0);
       SmartDashboard.putNumber("CDS Motor Speed", 0);
     }
   }
 
   public void ForwardCDS() {
```

Comment:
ForwardCDS() and ReverseCDS() are the same command and just toggle the direction of the speed. You should either do this with one Toggle() method, so it is clear that it is state-independent, or actually have ForwardCDS() and ReverseCDS() confirm that the motor is moving in the direction intended. 

For example, if you call ForwardCDS() twice in a row, it will go in opposite directions each time.